compiler/rustc_mir_build/src/thir/pattern/_match.rs

```diff
@@ -820,8 +820,8 @@ enum Constructor<'tcx> {
     Single,
     /// Enum variants.
     Variant(DefId),
-    /// Literal values.
-    ConstantValue(&'tcx ty::Const<'tcx>),
+    /// String literals
+    Str(&'tcx ty::Const<'tcx>),
```

Comment:
Okay so this type is indeed only used for exhaustiveness checking, that's why we do not need consts in here? Maybe add that to the type doccomment.

Reply:
Why are strings special here?